lib/rules/sort-destructure-keys.js

```diff
@@ -170,19 +144,76 @@ module.exports = {
     const sorter = createSorter(caseSensitive);
 
     return {
-      ObjectPattern(node) {
+      ObjectPattern(objectPatternNode) {
+        const scope = context.getScope();
+
+        function isReferencedByOtherProperties(id) {
```

Comment:
Can we make this a top-level function in the module?

Looks like we'll need to take `scope` as a  parameter as it's currently being closed over implicitly.